stackit/internal/services/iaas/network/resource.go

```diff
@@ -450,59 +601,129 @@ func toCreatePayload(ctx context.Context, model *Model) (*iaas.CreateNetworkPayl
+			Gateway:      iaas.NewNullableString(conversion.StringValueToPointer(model.IPv4Gateway)),
+			Prefix:       conversion.StringValueToPointer(model.IPv4Prefix),
+			PrefixLength: conversion.Int64ValueToPointer(model.IPv4PrefixLength),
+		}
 	}
 
 	labels, err := conversion.ToStringInterfaceMap(ctx, model.Labels)
 	if err != nil {
 		return nil, fmt.Errorf("converting to Go map: %w", err)
 	}
 
-	return &iaas.CreateNetworkPayload{
-		Name: conversion.StringValueToPointer(model.Name),
-		AddressFamily: &iaas.CreateNetworkAddressFamily{
-			Ipv4: &iaas.CreateNetworkIPv4Body{
-				PrefixLength: conversion.Int64ValueToPointer(model.IPv4PrefixLength),
-				Nameservers:  &modelNameservers,
-			},
-		},
+	payload := iaas.CreateNetworkPayload{
+		Name:   conversion.StringValueToPointer(model.Name),
 		Labels: &labels,
-	}, nil
+		Routed: conversion.BoolValueToPointer(model.Routed),
+	}
+
+	if addressFamily.Ipv6 != nil || addressFamily.Ipv4 != nil {
+		payload.AddressFamily = addressFamily
+	}
+
+	return &payload, nil
 }
 
-func toUpdatePayload(ctx context.Context, model *Model, currentLabels types.Map) (*iaas.PartialUpdateNetworkPayload, error) {
+func toUpdatePayload(ctx context.Context, model, stateModel *Model) (*iaas.PartialUpdateNetworkPayload, error) {
 	if model == nil {
 		return nil, fmt.Errorf("nil model")
 	}
+	addressFamily := &iaas.UpdateNetworkAddressFamily{}
 
-	modelNameservers := []string{}
-	for _, ns := range model.Nameservers.Elements() {
-		nameserverString, ok := ns.(types.String)
+	modelIPv6Nameservers := []string{}
+	for _, ipv6ns := range model.IPv6Nameservers.Elements() {
+		ipv6NameserverString, ok := ipv6ns.(types.String)
 		if !ok {
 			return nil, fmt.Errorf("type assertion failed")
 		}
-		modelNameservers = append(modelNameservers, nameserverString.ValueString())
+		modelIPv6Nameservers = append(modelIPv6Nameservers, ipv6NameserverString.ValueString())
 	}
 
+	if !model.IPv6Nameservers.IsNull() && len(model.IPv6Nameservers.Elements()) > 0 {
+		addressFamily.Ipv6 = &iaas.UpdateNetworkIPv6Body{
+			Nameservers: &modelIPv6Nameservers,
+			Gateway:     iaas.NewNullableString(conversion.StringValueToPointer(model.IPv6Gateway)),
+		}
+	}
+
+	modelIPv4Nameservers := []string{}
+	var modelIPv4List []attr.Value
+
+	if !(model.IPv4Nameservers.IsNull() || model.IPv4Nameservers.IsUnknown() || (model.IPv4Nameservers.Equal(stateModel.IPv4Nameservers) && !model.Nameservers.Equal(stateModel.Nameservers))) {
+		modelIPv4List = model.IPv4Nameservers.Elements()
+	} else {
+		modelIPv4List = model.Nameservers.Elements()
+	}
+	for _, ipv4ns := range modelIPv4List {
+		ipv4NameserverString, ok := ipv4ns.(types.String)
+		if !ok {
+			return nil, fmt.Errorf("type assertion failed")
+		}
+		modelIPv4Nameservers = append(modelIPv4Nameservers, ipv4NameserverString.ValueString())
+	}
+
+	if (!model.IPv4Nameservers.IsNull() && len(model.IPv4Nameservers.Elements()) > 0) || (!model.Nameservers.IsNull() && len(model.Nameservers.Elements()) > 0) {
```

Comment:
```suggestion
	if !model.IPv4Nameservers.IsNull() || !model.Nameservers.IsNull() {
```

Reply:
Same here, otherwise updating to an empty list would fail with an inconsistent result after apply